### PR TITLE
[client/rest]: add NEM rosetta account routes

### PR DESCRIPTION
**Pull request by Jaguar0625**
problem: NEM is missing rosetta account routes
solution: add them

### CODE REVIEW

File: client/rest/src/plugins/rosetta/nem/accountRoutes.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	register: (server, db, services) => {
+		const blockchainDescriptor = getBlockchainDescriptor(services.config);
+		const network = NetworkLocator.findByName(Network.NETWORKS, blockchainDescriptor.network);
+		const lookupCurrency = createLookupCurrencyFunction(services.proxy);
+		const getChainHeight = () => services.proxy.fetch('chain/height', json => json.height);
+		const getBlockIdentifier = height => services.proxy.fetch('local/block/at', undefined, {
+			method: 'POST',
+			headers: { 'Content-Type': 'application/json' },
+			body: JSON.stringify({ height })
+		}).then(blockInfo => new BlockIdentifier(Number(blockInfo.block.height), blockInfo.meta.hash));
```
Comment (Wayonb) [inline client/rest/src/plugins/rosetta/nem/accountRoutes.js:46]: Dont see a meta on the block 🤔 
```
curl http://127.0.0.1:7890/local/block/at --header 'Content-Type:application/json' --data '{ "height": 2 }' -s | jq
{
  "difficulty": 100000000000000,
  "txes": [],
  "block": {
    "timeStamp": 292549,
    "signature": "ba7cc40e62bfb180bc28b2f386d6595bbba35ed819185ee9dcddeaff66583c5e4810756aeadb62a218ce5081a5e847da4d83b538d3bf3a016c542892c3ec130e",
    "prevBlockHash": {
      "data": "f8c102ac15462c9f0e5629d0ea4f6fa0de80402763dbc48d946be2dda7c260ab"
    },
    "type": 1,
    "transactions": [],
    "version": -1744830463,
    "signer": "74375c15c6ce6bdbde59be88a069745a0de34444ea933f8c9f46ef407cf30196",
    "height": 2
  },
  "hash": "6cdf8d615a1db84ba07299e42077cc1e2c766e4e68d19458374f1bd355d50d3a"
}
```

File: client/rest/src/plugins/rosetta/nem/accountRoutes.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+		const getOperations = async unconfirmedTransactions => {
+			const mempoolTransactions = await Promise.all(unconfirmedTransactions.map(transaction =>
+				parser.parseTransactionAsRosettaTransaction(transaction.transaction, transaction.meta)));
+			return [].concat(...mempoolTransactions.map(transaction => transaction.operations));
+		};
+
+		const getAccountAmounts = async typedRequest => {
+			if (typedRequest.block_identifier)
+				throw RosettaErrorFactory.NOT_SUPPORTED_ERROR;
```
Comment (Wayonb) [inline client/rest/src/plugins/rosetta/nem/accountRoutes.js:67]: This is fine if not supporting historical node

**Review by Wayonb — APPROVED** [review]
👍🏾

File: client/rest/src/plugins/rosetta/nem/accountRoutes.js (hunk 2 above)
Comment (Jaguar0625) [reply client/rest/src/plugins/rosetta/nem/accountRoutes.js:67]: would leave this until/unless we add historical node support

**Review by Wayonb — APPROVED** [review]
👍🏾

**Pull request merged**